pkg/registry/client.go

```diff
@@ -209,74 +183,134 @@ func ClientOptPlainHTTP() ClientOption {
 	}
 }
 
-// ClientOptResolver returns a function that sets the resolver setting on a client options set
-func ClientOptResolver(resolver remotes.Resolver) ClientOption {
```

Comment:
This function is part of the public SDK interface for Helm. We can't remove it until we make a major version increment. This is documented in a [Helm Improvement Proposal](https://github.com/helm/community/blob/main/hips/hip-0004.md).

Reply:
Not seeing this happening with ORAS v2, containerd support has been removed from ORAS v2. See https://github.com/oras-project/oras-go/issues/64#issuecomment-1077845399.

Reply:
FWIW: This specific part of the Helm public SDK was only added in release [3.13.0](https://github.com/helm/helm/releases/tag/v3.13.0) by 80bc7df7820513e79fc24a1d4bbffd96ef6a9adf, it won't have seen much adoption yet. Perhaps there's a way to address the original issue's (#10623) request by allowing a way to specify custom HTTP headers? 

---

I'm not very familiar with golang, but it seems that [`ClientOptHTTPClient`](https://github.com/helm/helm/blob/v3.13.1/pkg/registry/client.go#L195) should satisfy #10623, no? I'm just trying to find more info to help move #12491 forward since this PR also seems to fix that issue.

Reply:
A middle ground here, might be to retain this `ClientOptResolver(resolver remotes.Resolver)`, but make it a noop (or make an error/panic)? The API would remain backwards compatible, just the behavior would (obviously) change.

Reply:
Isn't `panic`ing too aggressive? I would have the `NewClient()` return an error if `client.resolver != nil` evaluates to true

Reply:
@zregvart just checking if you saw the comment above

Reply:
@banjoh perhaps a better option would be to pass an error via the Client struct, as there is no `client.resolver` with this change, and I prefer not to reintroduce it. See [this diff](https://github.com/helm/helm/compare/498142f7035f1504f00fc16a71fd80cc010e8a79..05fe6feed3651b1497480f18bb0387e19525d4c1).